runtime/cmake/platform/toolcfg/gnu.cmake

```diff
@@ -20,7 +20,8 @@
 # SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
 ################################################################################
 
-list(APPEND OMR_PLATFORM_COMPILE_OPTIONS -O3 -g -fstack-protector)
+list(APPEND OMR_PLATFORM_COMPILE_OPTIONS -O3 -g -fstack-protector -Wreturn-type)
```

Comment:
This will still result in `-Wreturn-type` being given to `nasm`; see [OmrPlatform.cmake](https://github.com/eclipse/omr/blob/master/cmake/modules/OmrPlatform.cmake#L62).

Reply:
It should be fine. cmake tracks the different asm dialects. plain "ASM" is when we use the c compiler as an assembler.  Where as when we use nasm the language is `ASM_NASM`.

Reply:
Why would we want to use `-Wreturn-type` even for the 'ASM' language?
Wouldn't it be safer to add `-Wreturn-type` only to `OMR_PLATFORM_C_COMPILE_OPTIONS` and `OMR_PLATFORM_CXX_COMPILE_OPTIONS`?